### PR TITLE
autoDeserialize in KRead

### PR DESCRIPTION
**Pull request by gtrepta**
Can now do `krun --parser "cat" _.{json,kast,bin}`

### CODE REVIEW

**Review by ehildenb — DISMISSED** [review]
I'm approving, but please don't merge until @dwightguth also approves.

File: kernel/src/main/java/org/kframework/parser/KRead.java
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/parser/KRead.java:71]: can we just check that the first non-whitespace character in the input is '{'? instead of trying to parse it, failing, and then trying to parse it again differently? it will be a lot faster that way.

**Review by dwightguth — CHANGES_REQUESTED** [review]

**Review by ehildenb — APPROVED** [review]

File: kernel/src/main/java/org/kframework/parser/KRead.java
Comment (gtrepta) [reply kernel/src/main/java/org/kframework/parser/KRead.java:71]: Sure, that makes more sense.

**Review by dwightguth — APPROVED** [review]

**Pull request merged**
